### ISSUE TITLE
runtime require inconsistency

### ISSUE THREAD

**Issue by GordonSmith** [OpenNTF/dojo-webpack-plugin#67]
<edit - I initially thought this was an IE11 issue, but I can reproduce in chrome>

Opening the sample application test page fails when locale is set to en-gb (https://openntf.github.io/dojo-webpack-plugin-sample/test.html):
```
Error: Module not found: dojo/cldr/nls/en-gb/gregorian
   {
      [functions]: ,
      __proto__: { },
      description: "Module not found: dojo/cldr/nls/en-gb/gregorian",
      message: "Module not found: dojo/cldr/nls/en-gb/gregorian",
      name: "Error",
      stack: "Error: Module not found: dojo/cldr/nls/en-gb/gregorian
   at d (https://openntf.github.io/dojo-webpack-plugin-sample/release/bundle.js:1:895)
   at Anonymous function (https://openntf.github.io/dojo-webpack-plugin-sample/release/bundle.js:1:1294)
   at l (https://openntf.github.io/dojo-webpack-plugin-sample/release/bundle.js:1:1265)
   at i (https://openntf.github.io/dojo-webpack-plugin-sample/release/bundle.js:1:123)
   at Anonymous function (https://openntf.github.io/dojo-webpack-plugin-sample/release/bundle.js:1:102282)
   at l (https://openntf.github.io/dojo-webpack-plugin-sample/release/bundle.js:1:1265)
   at i (https://openntf.github.io/dojo-webpack-plugin-sample/release/bundle.js:1:123)
   at g (https://openntf.github.io/dojo-webpack-plugin-sample/release/bundle.js:1:102212)
   at Anonymous function (https://openntf.github.io/dojo-webpack-plugin-sample/release/bundle.js:1:102947)
   at s.forEach (https://openntf.github.io/dojo-webpack-plugin-sample/release/bundle.js:1:15942)"
   }
```

Steps to reproduce:
Add the following locale override to dojo config:
```
        locale: "en-gb",        
```
Rebuild and open.

<edit>
There are two issue going on, the missing locale is covered by:  https://github.com/OpenNTF/dojo-webpack-plugin/issues/69 and this issue will focus on the inconsistent behaviour of the runtime require call.

**Comment by GordonSmith** [comment]
FWIW I think its an issue with the webpack require implementation and the following "quick fix" in DojoAMDMainTemplate.runtime.js appears to work:

```
	function req(config, dependencies, callback) {
		try {
			return contextRequire(config, dependencies, callback, 0, req);
		} catch (e) {
			if (dependencies) {
				dependencies();
			}	
			return null;
		}	
	}
```

**Comment by chuckdumont** [comment]
Re-opening this because I don't believe we have it quite right yet.  Dojo doesn't throw an exception in the require call if a require dependency cannot be loaded, but it also does **not** call the require callback if any of the modules cannot be loaded and initialized.

Please create a new PR to fix this.  You could revert the change you made to `findModule()` so that it always throws again, and surround the call to `findModule()` in `contextRequire()` with a try/catch block and set a flag if an exception is caught, then only call the require callback if the flag has not been set.  You'll also need to update the unit test.  Thanks.

**Comment by GordonSmith** [comment]
I will do - (and I did just sanity check the above behavior and can confirm it is correct for my en-gb scenario).

FWIW I was basing my observed behavior on the "define" section.  Not the runtime require case!

I have opened a new issue:  https://github.com/OpenNTF/dojo-webpack-plugin/issues/69 which more accurately describes my original issue, and have edited this issue to reflect the issue we are currently fixing.
